crates/turborepo-lib/src/run/scope/filter.rs

```diff
@@ -641,13 +641,16 @@ mod test {
                     acc
                 });
 
-        let package_jsons = packages
+        let package_jsons = package_dirs
             .iter()
             .map(|package_path| {
                 let (_, name) = get_name(package_path);
                 (
                     turbo_root
-                        .join_unix_path(RelativeUnixPathBuf::new(**package_path).unwrap())
+                        .join_unix_path(
+                            RelativeUnixPathBuf::new(format!("{package_path}/package.json"))
```

Comment:
The keys to the package json map for the package graph builder should be paths to `package.json` files as this is what will get returned by a the `PackageManager::get_package_jsons`.

This caused the test fixture to cover up the logic errors in the implementation.